modules/core/src/Controller/Exception.php

```diff
@@ -38,6 +47,71 @@ public function __construct(
     }
 
 
+    /**
+     * Show Service Provider error.
+     *
+     * @param Request $request The request that lead to this login operation.
+     * @param string $code The error code
+     * @return \SimpleSAML\XHTML\Template  An HTML template
+     */
+    public function error(Request $request, string $code): Response
+    {
+        if ($code !== 'ERRORURL_CODE') {
+            Assert::oneOf($code, self::CODES);
+        }
+
+        $ts = $request->query->get('ts');
+        $rp = $request->query->get('rp');
+        $tid = $request->query->get('tid');
+        $ctx = $request->query->get('ctx');
+
+        if ($ts !== 'ERRORURL_TS') {
+            Assert::integerish($ts);
+            $ts = date(DATE_W3C, intval($ts));
+        } else {
+            $ts = null;
+        }
+
+        if ($rp !== 'ERRORURL_RP') {
+            $rp = urldecode($rp);
```

Comment:
So this urldecodes it twice I think since Symfony will already have done that in request->query->get()?

Reply:
I don't think Symfony does this, but I will install this patch and figure it out

Reply:
I'm surprised to find out that Symfony _does_ decode it! Thanks for pointing this out!